docs/cow-protocol/tutorials/arbitrate/autopilot.md

```diff
@@ -1,6 +1,150 @@
+    deactivate driver 1
+    driver 2->>autopilot: Proposed batch
+    deactivate driver 2
+
+    Note over autopilot: Pick winner
+    autopilot->>+driver 2: /reveal
+    driver 2->>-autopilot: Ethereum transaction data
+    autopilot->>+driver 2: /settle
+    driver 2->>+blockchain: Execute transaction
+    blockchain->>-driver 2: Transaction receipt
+    driver 2->>-autopilot: Transaction hash
+
+    autopilot->>database: Store auction data
+```
+
+[^barn]: There is technically also a second autopilot running on each network for testing purposes (in the _barn_ environment), but this isn't necessary for running the protocol.
+
+## Data collection
+
+The autopilot needs to have a complete overview of the protocol and of many other third-party protocols in order to build an [auction](/cow-protocol/reference/core/auctions/schema) that is as complete as possible.
+
+Much of the order data it needs is collected through the database, which is shared with the orderbook.
+The database stores the vast majority of available order information, including user signatures.
+
+Other information can only be retrieved onchain and is updated every time a new block is mined. For example, it needs to know from the protocol:
+
+- Which [pre-signatures](/cow-protocol/reference/core/signing-schemes#presign) have been set
+- If new [eth-flow orders](/cow-protocol/tutorials/cow-swap/native#eth-flow) have been created
+- Tracking which orders have been [invalidated](/cow-protocol/reference/contracts/core/settlement#invalidateorder) by the user
+- Detecting if a batch has been settled and it should prepare a new auction
+
+Retrieved information isn't limited to the CoW Protocol itself.
+The aution data also contains information on which liquidity is available to the solvers to settle an auction's orders.
+Solvers may use this liquidity information to help them out with finding a solution more quickly.
+Some liquidity sources are also used to provide a reference price of each token in an order;
+the reference price is used to normalize the value of the [surplus](/cow-protocol/reference/core/auctions/the-problem), since the surplus must be comparable for all orders and an order could use the most disparate ERC-20 tokens.
+
+Examples of liquidity sources that are tracked by the autopilot:
+- Uniswap (v2 and v3)
+- Balancer (Weighted, Weighted2Token, ComposableStable, ...)
+- SushiSwap
+- Swapr
+
+## Cutting auctions
+
+The autopilot constantly updates the list of orders that can be settled based on the data retrieved above.
+
+Orders that can't be settled are filtered out: these are expired orders, those with not enough balance, with missing approvals, or that use tokens that aren't supported by the protocol.
+
+The autopilot also checks that [ERC-1271](/cow-protocol/reference/core/signing-schemes#erc-1271) signatures are currently valid.
+
+This steps tries to guarantees that all available orders can be settled on CoW Protocol without further checks from the solvers.
```

Comment:
We filter orders partially in the `autopilot`. If a user only has 0.5 ETH we filter out all orders that sell more than that because there is absolutely no way they can be solved. (for partially fillable orders we only require the user to have at least 1 wei)
But if that user has 10 0.4 ETH orders we forward all of them to the solvers and let them figure out which one to prioritize.
So we filter out some orders but that does not yet guarantee that a solver can pick and solve any number of the orders without checks.

Reply:
I remove this line.
It feels like there should be an explanation here on why we prune the orders and we don't let the solvers get all of them regardless of these checks. (For example, a solver might be able to see that, after settling another order, a user would have enough balance to settle another one.)
I'm not sure what to write, however.

Reply:
> (For example, a solver might be able to see that, after settling an order, a user would have enough balance to settle another one.)

This case doesn't really work, though. All orders happen atomically so there is no before and after within the same block.

> I'm not sure what to write, however.

I would write something like the `autopilot` only filters out orders that are guaranteed to be unsettleable. All orders where it's ambiguous/unclear get forwarded to the solver to let it decide instead.